src/main/java/org/terasology/logic/health/event/FullHealthEvent.java

```diff
@@ -0,0 +1,36 @@
+/*
+ * Copyright 2019 MovingBlocks
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.terasology.logic.health.event;
+
+import org.terasology.entitySystem.entity.EntityRef;
+import org.terasology.entitySystem.event.Event;
+
+/**
+ * Event sent upon an entity reaching full health if previously on less than full health.
+ *
+ */
+public class FullHealthEvent implements Event {
```

Comment:
Might this be more descriptive as `OnFullHealthEvent` ?

Would also put it more in line with `OnHealedEvent` - although then I'd wonder if this would work better as `OnFullyHealedEvent` plus wonder what the implications of both triggering would be

Reply:
Renaming it as `OnFullyHealedEvent`, seems more apt for full health event.